lib/network_utils.ml

```diff
@@ -921,8 +922,31 @@ module Ovs = struct
 			| Some x, None -> ["--"; "set"; "bridge"; name; "mcast_snooping_enable=" ^ (string_of_bool x)]
 			| _ -> []
 		in
-		vsctl ~log:true (del_old_arg @ ["--"; "--may-exist"; "add-br"; name] @
-			vlan_arg @ mac_arg @ fail_mode_arg @ disable_in_band_arg @ external_id_arg @ vif_arg @ set_mac_table_size @ set_igmp_snooping)
+        let igmp_snooping_old =
+            let igmp_snooping_old' = 
+                try
+                    String.rtrim (vsctl ["--"; "get"; "bridge"; name; "mcast_snooping_enable"])
+                with _ -> "false"
+            in
+			if igmp_snooping_old' = "true" then
+				true
+			else
+				false
+        in
+        let creat_bridge_out = vsctl ~log:true (del_old_arg @ ["--"; "--may-exist"; "add-br"; name] @
+            vlan_arg @ mac_arg @ fail_mode_arg @ disable_in_band_arg @ external_id_arg @ vif_arg @ set_mac_table_size @ set_igmp_snooping)
+        in
+        begin
+            match igmp_snooping_old, igmp_snooping with
+                | false, Some true ->
+                    begin
+                        try
+                            ignore (call_script ~log_successful_output:true inject_igmp_query ["--detach"; "--max-resp-time"; "50"; "bridge"; name])
```

Comment:
I prefer this to be a separate function in network_utils as well, just to call the inject script, so that these utils functions remain plain and simple. The `Bridge.create` function in network_server.ml can do the orchestration.